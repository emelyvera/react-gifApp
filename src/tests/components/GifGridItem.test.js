import React from 'react';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas del componente <GifGridItem />', () => {

  const title = 'Titulo';
  const url = 'https://local';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should mostrar el componente correctamente <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should de tener un parrafo con el title', () => {

    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);

  });

  test('should tener una img igual al url y alt de los props', () => {

    const img = wrapper.find('img');
    // console.log( img.props() );
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  });

  test('should que el div tenga el animate__findeIn', () => {
    
    const div = wrapper.find('div');
    
    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
  });
  
});

