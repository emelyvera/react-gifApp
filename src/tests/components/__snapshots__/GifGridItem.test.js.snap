// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGridItem /> should mostrar el componente correctamente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://local"
  />
  <p>
    Titulo
  </p>
</div>
`;
